### PR TITLE
chore: merge 3.6 to main

### PR DESCRIPTION
**Pull request by Aflynn50**
Merge up 3.6 into main. This merge had plenty of conflicts. They are listed below with how they were resolved.

Content in this merge up:
- #17540 from Aflynn50
  - #17538 from Aflynn50
    - #17513 from Aflynn50
      Conflicts resolved fully in favour of main:
      - apiserver/facades/client/modelupgrader/upgrader_test.go
      - internal/upgrades/upgradevalidation/migrate_test.go
      - internal/upgrades/upgradevalidation/upgrade_test.go
      - internal/upgrades/upgradevalidation/validation_test.go
      - core/base/supported.go
    - #17485 from hpidcock
      Conflicts resolved to keep changes:
      - internal/worker/migrationminion/manifold.go
      - internal/worker/migrationminion/worker.go
      - internal/worker/migrationminion/worker_test.go
    - #17510 from SimonRichardson
    - #17497 from nvinuesa
      Conflicts resolved fully in favour of main:
      - cmd/juju/ssh/debughooks_test.go
      - state/application_test.go 
    - #17502 from ycliuhw/fix
      Conflicts resolved fully in favour of main:
      - internal/worker/uniter/uniter.go
    - #17484 from hpidcock
  - #17503 from barrettj12

### CODE REVIEW

**Review by SimonRichardson — COMMENTED** [review]
My changes and the changes I approved look sane.

**Review by jack-w-shaw — COMMENTED** [review]
Looks sensible to me too

You're defo right to resolve conflict resulting from https://github.com/juju/juju/pull/17513 in favour of main

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -2,10 +2,10 @@ name: "Docs"
 on:
   push:
     branches:
-      - '3.3'
+      - '3.5'
   pull_request:
     branches:
-      - '3.3'
+      - '3.5'
```
Comment (jack-w-shaw) [inline .github/workflows/docs.yml:8]: Should these be rolled up to main?

**Review by jack-w-shaw — APPROVED** [review]

**Pull request merged**